### ISSUE TITLE
Slow response (API calls) when operating through VPN

### ISSUE THREAD

**Issue by jacksongoode** [jpochyla/psst#278]
When using a VPN (in my case Mullwad on Fedora 35), Psst is considerably slower in requests and the UI blocks as a result. In my tests, I don't latency could really be a factor here as I only go from ~9ms to ~20ms with VPN off to on. I'm not sure what the hang could be - perhaps it's at Spotify's end? - But definitely something to note?

**Comment by kingosticks** [comment]
Are you saying that using psst while connected to your VPN is slower than using the official client while connected to your VPN? Or that using psst while connected to your VPN is closer than using psst while not using your VPN? It's always going to be slower through a VPN, seemingly twice as slow by your measurement?How are you measuring the latency to Spotify's servers?

**Comment by jacksongoode** [comment]
> Are you saying that using psst while connected to your VPN is slower than using the official client while connected to your VPN? Or that using psst while connected to your VPN is closer than using psst while not using your VPN? It's always going to be slower through a VPN, seemingly twice as slow by your measurement - which sounds reasonable. You are, after all, forcibly sending your traffic the wrong way.

Yes, it makes sense that there would be some delay in requests - but it really should only be subtle, I can't imagine it would exceed 100ms right? The tested VPN location is within 50km of my location. However, the response time delay increase is in the factor of 4 or 5 times longer. This, to me, seems like either Spotify isn't happy with the request origin (the VPN IP?) or there's some hang up within Psst.

> How are you measuring the latency to Spotify's servers?
Pinging `spotify.com`

**Comment by jpochyla** [comment]
That's, uhh... strange? No idea where to look at the moment :|

**Comment by kingosticks** [comment]
@jacksongoode can you provide which access point is being used in both instances. Psst logs a trace message with this info.
